File: COS333/Practical6/s20526254.pl
#!/usr/bin/perl
use strict;
use warnings;

if (@ARGV != 2) {
  die "Usage: perl script.pl <filename> <word>\n";
}

my ($filename, $searchWord) = @ARGV;

open my $fh, '<', $filename or die "Could not open file '$filename' $!";

my $content = '';
while (my $line = <$fh>) {
  $content .= $line;
}
close $fh;

$content = lc($content);
print "Original File Contents:\n";
print "-----------------------\n";
print "$content";
print "-----------------------\n";
$content =~ s/[^a-z]//g;
print "\nFile Contents after conversion: $content\n";

my $reversedContent = reverse $content;
my $matches = () = $reversedContent =~ /$searchWord/g;

print "Search Word: $searchWord\n";
print "Matches: $matches\n";

File: COS333/Practical6/s20526254.php
<?php
  $filename = $argv[1];
  $searchWord = $argv[2];

  if (!file_exists($filename)) {
    echo "File not found.\n";
    exit(1);
  }

  $content = file_get_contents($filename);
  echo "Original File Contents:\n";
  echo "-----------------------\n";
  echo "$content";
  echo "-----------------------\n";
  $content = strtolower($content);
  $content = preg_replace('/[^a-z]/', '', $content);
  echo "\nFile Contents after conversion: $content\n";

  $reversedContent = strrev($content);
  $matches = substr_count($reversedContent, $searchWord);

  echo "Search Word: $searchWord\n";
  echo "Matches: $matches\n";

  $output = "<!DOCTYPE html>
  <html>
  <head>
      <title>Word Count</title>
  </head>
  <body>
      <p>Matches: $matches</p>
  </body>
  </html>\n";
  
  $outputFile = 'output.html';
  file_put_contents($outputFile, $output);
  echo "\nOutput written to $outputFile\n";

  exit(0);
?>
